### ISSUE TITLE
Carthage - Unable to install with Xcode 11.2.1 (11B500)

### ISSUE THREAD

**Issue by ozgur** [urbanairship/ios-library#231]
❗For how-to inquiries involving Airship functionality or use cases, please
contact (support)[https://support.airship.com/].

# Preliminary Info

### What are the versions of any relevant development tools you are using?

# Report

### What unexpected behavior are you seeing?

```
carthage install ios-library --platform iOS
```
I am unable to install AirshipKit. It says:
```
*** Downloading ios-library.framework binary at "12.1.0"
***  Skipped installing ios-library.framework binary due to the error:
	"Incompatible Swift version - framework was built with 5.1 (swiftlang-1100.0.270.13 clang-1100.0.33.7) and the local version is 5.1.2 (swiftlang-1100.0.278 clang-1100.0.33.9)."
``` 

### Do you have logging for the issue?

```
    Falling back to building from the source
*** Building scheme "AirshipAppExtensions" in Airship.xcworkspace
*** Building scheme "AirshipDebugKit" in Airship.xcworkspace
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace /Users/ozgur/Developer/test-ios/Test/Carthage/Checkouts/ios-library/Airship.xcworkspace -scheme AirshipDebugKit -configuration Release -derivedDataPath /Users/ozgur/Library/Caches/org.carthage.CarthageKit/DerivedData/11.2.1_11B500/ios-library/12.1.0 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/ly/76vl5lmn2ln560j9d7wpq9vm0000gp/T/ios-library SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/ozgur/Developer/test-ios/Test/Carthage/Checkouts/ios-library)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/ly/76vl5lmn2ln560j9d7wpq9vm0000gp/T/carthage-xcodebuild.2egKAW.log
```
Here is the last 10 lines of the log:
```
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/AirshipKit.h"
        ^
/Users/ozgur/Library/Caches/org.carthage.CarthageKit/DerivedData/11.2.1_11B500/ios-library/12.1.0/Build/Intermediates.noindex/ArchiveIntermediates/AirshipDebugKit/BuildProductsPath/Release-iphoneos/AirshipKit.framework/Headers/AirshipKit.h:10:9: note: in file included from /Users/ozgur/Library/Caches/org.carthage.CarthageKit/DerivedData/11.2.1_11B500/ios-library/12.1.0/Build/Intermediates.noindex/ArchiveIntermediates/AirshipDebugKit/BuildProductsPath/Release-iphoneos/AirshipKit.framework/Headers/AirshipKit.h:10:
#import "AirshipLib.h"
        ^
/Users/ozgur/Library/Caches/org.carthage.CarthageKit/DerivedData/11.2.1_11B500/ios-library/12.1.0/Build/Intermediates.noindex/ArchiveIntermediates/AirshipDebugKit/BuildProductsPath/Release-iphoneos/AirshipKit.framework/Headers/AirshipLib.h:148:9: error: 'UIWebView+UAAdditions.h' file not found
#import "UIWebView+UAAdditions.h"
        ^
/Users/ozgur/Developer/test-ios/Test/Carthage/Checkouts/ios-library/AirshipDebugKit/AirshipDebugKit/Device-Info/AddNamedUserTableViewController.swift:4:8: error: could not build Objective-C module 'AirshipKit'
import AirshipKit
       ^

** ARCHIVE FAILED **


The following build commands failed:
	CompileSwift normal arm64
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
```

**Comment by crow** [comment]
@ozgur Thanks for reporting, looking into it

**Comment by crow** [comment]
if you use the `--no-use-binaries` option what's your output? Could you please also provide your cartfile?

**Comment by crow** [comment]
@ozgur I'm fairly certain the issue is that the pre-build binary we supply is built the version of Swift available at the time of the release. If you use a newer version of Swift, `--no-use-binaries` is necessary so your machine with the newest Swift version re-builds the frameworks.

**Comment by ozgur** [comment]
I've just run the following command:
```
carthage update ios-library --platform iOS --no-use-binaries
```
Cartfile:
```
github "urbanairship/ios-library" == 12.1.0
```
and the error log is:
[carthage-xcodebuild.6XRY4Z.log](https://github.com/urbanairship/ios-library/files/3918033/carthage-xcodebuild.6XRY4Z.log)

**Comment by crow** [comment]
I'm able to run the following commands on my machine without any issue:
`carthage update ios-library --platform iOS --no-use-binaries`
cartfile:
`github "urbanairship/ios-library" == 12.1.0`

Could you give me the output when you run the following command?:
`/usr/bin/xcodebuild -version
`

**Comment by ozgur** [comment]
```
Xcode 11.2.1
Build version 11B500
```

**Comment by crow** [comment]
Looks like it's probably a caching issue. I'd try deleting the cache via:
`rm -rf ~/Library/Caches/org.carthage.CarthageKit`

**Comment by ozgur** [comment]
Tried again but no lock. Anyway, I'll have my friend build it on their own laptop to make sure if I am doing sth wrong. 

In the meantime, do you have plans on creating a new release built with Swift 5.1.2?

**Comment by gamesbydre** [comment]
I'm also unable to update using Xcode 11.2.1 (11B500).
I receive the same build error reported.

I deleted the cache, updated with --no-use-binaries, but the build fails during the AirshipDebugKit scheme.

`carthage update --platform ios,tvos --no-use-binaries`

```
*** Building scheme "AirshipAppExtensions" in Airship.xcworkspace
*** Building scheme "AirshipDebugKit" in Airship.xcworkspace
Build Failed
	Task failed with exit code 65:
        ...
This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/nm/zgkttgwj24g_gprbbjwy8zcw0000gn/T/carthage-xcodebuild.ngb2Jr.log
```

**Comment by rlepinski** [comment]
I believe this is fixed in 12.1.1

**Comment by ozgur** [comment]
Thanks. I was able to upgrade to 13.0.0.
